dev

```diff
@@ -4,26 +4,51 @@ set -e
 
 DIR=$(pwd)
 
+# network to connect multiple engine instances
+TENDERMINT_NETWORK="mesg-tendermint"
+
 # mesg config variables
 MESG_NAME=${MESG_NAME:-engine}
 MESG_PATH=${MESG_PATH:-$HOME/.mesg}
 MESG_LOG_FORMAT=${MESG_LOG_FORMAT:-text}
 MESG_LOG_FORCECOLORS=${MESG_LOG_FORCECOLORS:-false}
 MESG_LOG_LEVEL=${MESG_LOG_LEVEL:-debug}
+MESG_SERVER_PORT=${MESG_SERVER_PORT:-50052}
+MESG_TENDERMINT_PATH=${MESG_TENDERMINT_PATH:-"tendermint"}
+
+TENDERMINT_NETWORK_ALIAS=${TENDERMINT_NETWORK_ALIAS:-$MESG_NAME}
+TENDERMINT_VALIDATOR_NAME=${TENDERMINT_VALIDATOR_NAME:-validator}
+TENDERMINT_VALIDATOR_PORT=${TENDERMINT_VALIDATOR_PORT:-26656}
+TENDERMINT_VALIDATOR_PATH=${TENDERMINT_VALIDATOR_PATH:-".tendermint-validator"}
+
+# Init and read validator identity
+eval $(go run internal/tools/gen_tm_identity.go -name=$TENDERMINT_VALIDATOR_NAME -port=$TENDERMINT_VALIDATOR_PORT -path=$TENDERMINT_VALIDATOR_PATH)
+MESG_TENDERMINT_P2P_SEEDS=${MESG_TENDERMINT_P2P_SEEDS:-$NODE_PUBKEY}
+MESG_TENDERMINT_VALIDATORPUBKEY=${MESG_TENDERMINT_VALIDATORPUBKEY:-$VALIDATOR_PUBKEY}
+
+# Setup the validator private keys
+if [[ $* == *--validator* ]]; then
```

Comment:
@krhubert if you know a better way to read flag but definitely not blocking

Reply:
yes, there is `getopts` tool (one of many examples https://sookocheff.com/post/bash/parsing-bash-script-arguments-with-shopts/)